CloudyCar/app/src/main/java/com/cloudycrew/cloudycar/createrequest/CreateRequestActivity.java

```diff
@@ -3,50 +3,75 @@
-import android.support.v7.widget.CardView;
+import android.widget.EditText;
 import android.widget.TextView;
 
 import com.cloudycrew.cloudycar.BaseActivity;
 import com.cloudycrew.cloudycar.R;
+import com.cloudycrew.cloudycar.SummaryActivity;
 import com.cloudycrew.cloudycar.models.Route;
 
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
 /**
  * Created by George on 2016-11-05.
  */
 
 public class CreateRequestActivity extends BaseActivity implements ICreateRequestView {
+    @BindView(R.id.set_price)
+    protected EditText userPrice;
+    @BindView(R.id.suggested_price)
+    protected TextView suggestedPrice;
+    @BindView(R.id.submit_ride_request)
+    protected TextView submit;
+    @BindView(R.id.display_route_start)
+    protected TextView startText;
+    @BindView(R.id.display_route_end)
+    protected TextView endText;
+
     private CreateRequestController createRequestController;
+    private Route userRoute;
+    private double price;
+
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_create_request);
-        TextView textView = (TextView) findViewById(R.id.pricetextview);
-        textView.setCompoundDrawablesWithIntrinsicBounds(R.drawable.ic_attach_money_black_24dp, 0, 0, 0);
-
-        this.getSupportActionBar().setTitle(R.string.request_create_header);
+        ButterKnife.bind(this);
         resolveDependencies();
 
-        TextView submit = (TextView)this.findViewById(R.id.submit_ride_request);
-        TextView startText = (TextView)this.findViewById(R.id.display_route_start);
-        TextView endText = (TextView)this.findViewById(R.id.display_route_end);
-        Route userRoute = (Route) getIntent().getExtras().getSerializable("route");
-        Resources resouces = getResources();
-        String startToDisplay = String.format(resouces.getString(R.string.write_start_location),userRoute.getStartingPoint());
-        String endToDisplay = String.format(resouces.getString(R.string.write_end_location),userRoute.getEndingPoint());
-        startText.setText(startToDisplay);
-        endText.setText(endToDisplay);
+        userRoute = (Route) getIntent().getExtras().getSerializable("route");
+
+        this.getSupportActionBar().setTitle(R.string.request_create_header);
+        suggestedPrice.setCompoundDrawablesWithIntrinsicBounds(R.drawable.ic_attach_money_black_24dp, 0, 0, 0);
+        formatRouteDisplay(startText, endText);
 
         submit.setOnClickListener(v -> {
             //Submit request to server
+            if(userPrice.getText().length()>0){
+                price = Double.parseDouble(userPrice.getText().toString());
+            }else{
+                price = Double.parseDouble(suggestedPrice.getText().toString());
+            }
+            createRequestController.saveRequest(userRoute,price);
+            Intent intent = new Intent(this, SummaryActivity.class);
```

Comment:
saveRequest actually happens asynchronously so leaving the request activity before the request is created is a little sketchy. The way I designed the RequestController is mostly at fault for this though. I think if i switched the request controller to execute the request synchronously and have the CreateRequestController handle executing the request async, then the CreateRequestController can call a method in the view when the Request was successfully upserted (similarly to the SignUpController & Activity). I can make a change in the controller so it works like that.